### PR TITLE
Se feedback sprint 3&4

### PR DESCRIPTION
**Pull request by breijm**
Grading has been completed for both Sprints. New grading policies as discussed in the meeting have been applied. Questions and comments are welcome.

### CODE REVIEW

File: se_feedback/sprint-3/feedback_sprint_3_all.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| Architecture Design Document (ADD) |  8    |
+
+|                     | Score |
+|---------------------|-------|
+| Code Change Quality |  7    |
+
+| Code Readability | Score |
+|------------------|-------|
+| Formatting       | 10    |
+| Naming           |  8    |
```
Comment (Exclaminator) [inline se_feedback/sprint-3/feedback_sprint_3_all.md]: Could you give some examples of sub-optimal naming in the codebase?

**Comment by breijm** [comment]
@Exclaminator No problem

attribute naming inconsistency (c_style or camelCase)
**peer_comments** (Options.ts - line 12)
As opposed to **isDebug** (Logger.ts - line 11)

Capitalised function names?
Status.ts
line 14 - public static get NAME()
line 26 - public static get MESSAGE()

File: se_feedback/sprint-4/feedback_sprint_4_all.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+| Code Review            |  10    |
+
+##Notes
+* ADD
+	* Good class diagram
+	* A sequence diagram of how the user interacts with the options would be useful to add
+	* A sequence diagram of how the MainController interacts with the ContentController would make the textual explanation clearer
+	* A sequence diagram of how data is collected while a user is browsing a pr would make the textual explanation clearer
+	* Why is the Options class a Singleton?
+	* A flow diagram of how the `ContentController` deals with `ElementSelectionBinding` objects would be useful
```
Comment (Exclaminator) [inline se_feedback/sprint-4/feedback_sprint_4_all.md]: We are assuming that flow diagram and flow chart are used interchangably?

**Comment by breijm** [comment]
@Exclaminator 

Short answer: yes
Long answer: Any diagram/chart that shows a sequence/flow of events can be used

File: se_feedback/sprint-4/feedback_sprint_4_all.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	* Should also be able to remove an observer for clean code purposes
+* `DoNotWatchOptions` seems incomplete but there are no To Do tags
+	* Hardcoded pairs of events
+		* Create a settings file or object that can hold these
+* `ContentController`
+	* Consider moving large lists to a separate class or settings file
+	* `hookToDOM`
+		* Lines 142 / 149 - 153, these if conditions seems like good candidates for functions in the `DoNotWatchOptions` class
+* Event and Element IDs are magic number constants hidden in Constructors
+	* You have a documented table of these IDs, once this is finalised I suggest looking into putting these as consts in files so that the information is available in one place in the code as well
```
Comment (MathiasMeuleman) [inline se_feedback/sprint-4/feedback_sprint_4_all.md]: We like the idea of changing the magic numbers to some TypeSript equivalent of public static finals, but would you think this file should contain public static final ints or public static final EventIDs/ElementIDs

**Comment by breijm** [comment]
@MathiasMeuleman I'd pick the one that is easiest to modify (if that's ever needed). Say you change the EventID / ElementID classes do you want to have to change these constants as well? It's up to you.

**Comment by mdingjan** [comment]
I've read the feedback and understand it.
We've read the feedback points one by one and discussed them on monday morning with the team.
We're still processing it, but this is ready for merging.

**Comment by MathiasMeuleman** [comment]
Yes, I will merge this. If any issues arise, we will let you know

**Pull request merged**